core/domain/classifier_services.py

```diff
@@ -127,6 +129,115 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+def create_classifier_training_jobs(exploration, state_names):
+    """Creates ClassifierTrainingJobModel instances for all the state names
+    passed into the function.
+
+    Args:
+        exploration: Exploration. The Exploration domain object.
+        state_names: list(str). List of state names.
+    """
+    job_dicts_list = []
+    exp_id = exploration.id
+    exp_version = exploration.version
+    for state_name in state_names:
+        state = exploration.states[state_name]
+        training_data = state.get_training_data()
+        interaction_id = state.interaction.id
+        algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+            interaction_id]['algorithm_id']
+
+        # Validate the job.
+        dummy_classifier_training_job = classifier_domain.ClassifierTrainingJob(
+            'job_id_dummy', algorithm_id, interaction_id, exp_id, exp_version,
+            state_name, feconf.TRAINING_JOB_STATUS_NEW, training_data)
+        dummy_classifier_training_job.validate()
+
+        job_dicts_list.append({
+            'algorithm_id': algorithm_id,
+            'interaction_id': interaction_id,
+            'exp_id': exp_id,
+            'exp_version': exp_version,
+            'state_name': state_name,
+            'training_data': training_data,
+            'status': feconf.TRAINING_JOB_STATUS_NEW
+        })
+
+    # Create all the classifier training jobs.
+    job_ids = classifier_models.ClassifierTrainingJobModel.create_multi(
+        job_dicts_list)
+
+    # Create mapping for each job.
+    job_exploration_mappings = []
+    for job_id_index, job_id in enumerate(job_ids):
+        job_exploration_mapping = (
+            classifier_domain.TrainingJobExplorationMapping(
+                job_dicts_list[job_id_index]['exp_id'],
+                job_dicts_list[job_id_index]['exp_version'],
+                job_dicts_list[job_id_index]['state_name'],
+                job_id))
+        job_exploration_mapping.validate()
+        job_exploration_mappings.append(job_exploration_mapping)
```

Comment:
@prasanna08 We cannot append domain objects in the above part because we cannot create job domain objects until we know their job_ids, which we dont at that time. Its not the same with the mapping domain class though. Sean suggested it'd be better to pass domain objects instead of dicts here.